cli/generators/crud/templates/service.js

```diff
@@ -1,67 +1,43 @@
-const <%= pluralizedName %> = [];
+import mongodbConnection from './../database/mongodb';
+import <%= ucName %> from './<%= name %>.schema';
 
 class <%= ucName %>Service {
-  constructor() {
-    this.nextId = 0;
-
-    // Seed with some dummy data.
-    for (let i = 1; i <= 10; i++) {
-      var id = this._generateNextId();
-      <%= pluralizedName %>.push({
-        id: id,
-        name: `Test Name ${id}`
-      });
-    }
-  }
-
   findAll(cb) {
-    let all<%= pluralizedUcName %> = <%= pluralizedName %>;
-
-    return cb(null, all<%= pluralizedUcName %>);
+    <%= ucName %>.find()
+      .then(<%= pluralizedName %> => cb(null, <%= pluralizedName %>))
+      .catch(err => cb('Unable to retrieve <%= pluralizedName %>.'));
   }
 
   findById(id, cb) {
-    const <%= name %> = <%= pluralizedName %>.find(<%= name %> => <%= name %>.id == id);
-
-    return cb(null, <%= name %>);
+    <%= ucName %>.findById(id)
+      .then(<%= name %> => cb(null, <%= name %>))
+      .catch(err => cb('Unable to find <%= name %>.'));
   }
 
-  create(<%= name %>, cb) {
-    <%= name %>.id = this._generateNextId();
-    <%= pluralizedName %>.push(<%= name %>);
-
+  create(data, cb) {
+    let <%= name %> = new <%= ucName %>(data);
+    <%= name %>.save();
     return cb(null, <%= name %>);
   }
 
   update(id, data, cb) {
-    // Finding the index is only for memory storage.
-    const indexOf<%= ucName %>ToUpdate = <%= pluralizedName %>.findIndex((<%= name %>) => { return <%= name %>.id == id });
-    if (indexOf<%= ucName %>ToUpdate === -1)
-      return cb(`The <%= name %> doesn't exist.`);
-
-    // Get the "stored" record in memory and update with the new data.
-    let <%= name %> = <%= pluralizedName %>.find(<%= name %> => <%= name %>.id == id);
-    Object.assign(<%= name %>, data);
-
-    // Put the updated record on memory.
-    <%= pluralizedName %>.splice(indexOf<%= ucName %>ToUpdate, 1, <%= name %>);
-
-    return cb(null, <%= name %>);
+    delete data.id;
+
+    <%= ucName %>.findByIdAndUpdate(id, data, { new: true }) // To return the modified document rather than the original.
```

Comment:
Like your proposal.

Unfortunately, no. It causes what is described. "True" in this particular API means that it will return the updated value. Is a really really confusing signature. http://mongoosejs.com/docs/api.html#model_Model.findByIdAndUpdate